src/main/java/duke/commands/Command.java

```diff
@@ -0,0 +1,74 @@
+package duke.commands;
+
+import duke.exceptions.DukeException;
+import duke.storage.TaskList;
+import duke.ui.Ui;
+
+/**
+ * The Command class defines a command that can be used in the Duke application.
+ *
+ * @author Ryan NgWH
+ */
+public abstract class Command {
+    /**
+     * Flag indicating if command is an exit command
+     */
+    private boolean isExit;
+
+    /**
+     * Creates a new Command
+     *
+     * @param isExit Flag if command is an exist command
+     */
+    protected Command(boolean isExit) {
+        this.isExit = isExit;
+    }
+
+    /**
+     * Check if the command is an exit command
+     *
+     * @return True if the command is an exist command, false otherwise
+     */
+    public boolean isExit() {
+        return this.isExit;
+    }
+
+    /**
+     * Executes the command
+     *
+     * @param taskList Tasklist used for the command
+     * @param ui       UI used for the command
+     */
+    public abstract void execute(TaskList taskList, Ui ui) throws DukeException;
+
+    /**
+     * Executes the command
+     *
+     * @param taskList Tasklist used for the command
+     *
+     * @return String containing the output of the command
+     */
+    public abstract String execute(TaskList taskList) throws DukeException;
```

Comment:
I feel like overloading 'execute' may not be very representative of the functionality. Perhaps a name like getString or getOutputString would be more accurate

Reply:
Good point. Will consider your alternatives!